### ISSUE TITLE
Onedrive Sync fails in ios11

### ISSUE THREAD

**Issue by mmelasi** [loledeck/librairie.cubesoft.fr#3]
using an ipad pro 12” in ios11.
I recently purchased this app (in app with one drive sync) and allowed it full access to my onedrive account.  it will not sync any books from there.  It works fine with dropbox, but i have limited space there.  Is there a fix?  All my ebooks are epubs, in a root level folder called books. This is basically my Calibre library syncd to onedrive.   Help.

**Comment by loledeck** [comment]
Hello,

Could you please give me the email address you’re using for your OneDrive account ? It will allow me to check the import log on the server to see what the error was.
Thanks.

Best,
Laurent


Le 3 février 2018 à 17:32:59, mmelasi (notifications@github.com) a écrit:

using an ipad pro 12” in ios11.
I recently purchased this app (in app with one drive sync) and allowed it full access to my onedrive account. it will not sync any books from there. It works fine with dropbox, but i have limited space there. Is there a fix? All my ebooks are epubs, in a root level folder called books. This is basically my Calibre library syncd to onedrive. Help.

**Comment by mmelasi** [comment]
mmelasi@gmail.com

**Comment by loledeck** [comment]
I see 103 ebboks in your OneDrive account.

If it is the number you were expecting then it’s just a mater of refreshing your library on your iPad.
Go to the library view and pull down the view until you see the spinning wheel and let go. A few seconds/minutes later your ebooks should start to appear.

If it is not the case, unlink your account (by going to settings) and restart the connection with your OneDrive account.

Hope this helps.

Best,
Laurent


Le 3 février 2018 à 17:54:24, mmelasi (notifications@github.com) a écrit:

mmelasi@gmail.com

**Comment by mmelasi** [comment]
no, they aren’t showing up in the app.    i tried unlinking, resetting the database, reinstalling the app, etc.    I also refreshed several,times and let it sit for a couple hours.    it’s not working right.
  

  
  

  
  
>

**Comment by loledeck** [comment]
Ok, I made a mistake. The 103 ebooks were linked to your Dropbox account and your OneDrive had none.
I have run the import again from scratch and it found 17 ebooks. Those are now linked to your OneDrive account and should be visible by refreshing.

Sorry for the confusion.

Best,
Laurent


Le 3 février 2018 à 18:03:11, mmelasi (notifications@github.com) a écrit:




no, they aren’t showing up in the app. i tried unlinking, resetting the database, reinstalling the app, etc. I also refreshed several,times and let it sit for a couple hours. it’s not working right.







>

**Comment by mmelasi** [comment]
Yes, whatever you did, they show up now, so thank you.  However the sync feature is very flawed.  There is significant delay....for example i removed more than half the books from onedrive and added 2 new ones about 20 minutes ago and refreshed the app.  The books are not updated in the app to reflect this.  I have hundreds of books, so i suspect this is a fail and regret making the purchase based on this performance.

**Comment by loledeck** [comment]
Hello,

I’ve just checked your account again (not done anything, just checked) and I see that you now have 1915 ebooks synced with your OneDrive account.
If you refresh the app you should be able to see them.

Reading your last response, I think you may be a little confused about how the sync work:

- Once an account is linked with the app, the sync is triggered by refreshing the view in the app
- the app will then scan the changes in your OneDrive account to detect the file changes since the last sync
- if any file changed is an ebook in a supported format, the app will add it to your library. *This may take a while* as every single book has to be parsed, query different databases to find relevant information about it and then sync it back with your device
- None of the ebooks currently in your library are removed from it by the sync operation. If you want to get rid of an ebook (in the app only, we will never ever delete a file from your cloud account) you have to left swipe on it in the library view and tap the « delete » button.

The goal of the sync feature is not to reflect in a timely manner the content of your file storage cloud account. It is to let you easily import large quantities of ebooks in a reasonable amount of time and with minimum effort.

Best,
Laurent 

Le 3 février 2018 à 19:38:34, mmelasi (notifications@github.com) a écrit:

Yes, whatever you did, they show up now, so thank you. However the sync feature is very flawed. There is significant delay....for example i removed more than half the books from onedrive and added 2 new ones about 20 minutes ago and refreshed the app. The books are not updated in the app to reflect this. I have hundreds of books, so i suspect this is a fail and regret making the purchase based on this performance.

**Comment by mmelasi** [comment]
Ok yes I see it’s working as you’ve described.    Thanks for all the prompt follow ups!
  

  
  

  
  
>
